v-next/hardhat/src/internal/builtin-plugins/solidity-test/reporter.ts

```diff
@@ -117,27 +122,60 @@ export async function* testReporter(
     for (const [index, failure] of failures.entries()) {
       yield `\n${chalk.bold(chalk.red(`Failure (${index + 1})`))}: ${failure.name}\n`;
 
-      if (failure.decodedLogs !== undefined && failure.decodedLogs.length > 0) {
-        yield `Decoded Logs${chalk.grey(`: ${failure.decodedLogs.join("\n")}\n`)}`;
+      const stackTrace = failure.stackTrace();
+      if (
+        stackTrace !== undefined &&
+        stackTrace !== null &&
+        stackTrace.length > 0
+      ) {
+        const stackTraceMessage = getMessageFromLastStackTraceEntry(
+          stackTrace[stackTrace.length - 1],
+        );
+
+        const stackTraceStack: string[] = [];
+        for (const entry of stackTrace.reverse()) {
+          const callsite = encodeStackTraceEntry(entry);
+          if (callsite !== undefined) {
+            stackTraceStack.push(`  at ${callsite.toString()}`);
+          }
+        }
+
+        if (stackTraceMessage !== undefined || stackTraceStack.length > 0) {
+          yield `${stackTraceMessage ?? "Stack trace:"}\n${chalk.grey(stackTraceStack.join("\n"))}\n`;
+        }
+      }
+
+      if (
+        failure.decodedLogs !== undefined &&
+        failure.decodedLogs !== null &&
+        failure.decodedLogs.length > 0
+      ) {
+        yield `Decoded Logs:\n${chalk.grey(failure.decodedLogs.map((log) => `  ${log}`).join("\n"))}\n`;
       }
 
-      if (failure.reason !== undefined && failure.reason !== "") {
-        yield `Reason${chalk.grey(`: ${failure.reason}\n`)}`;
+      if (
+        failure.reason !== undefined &&
+        failure.reason !== null &&
+        failure.reason !== ""
+      ) {
+        yield `Reason:\n${chalk.grey(`  ${failure.reason}`)}\n`;
       }
 
-      if (failure.counterexample !== undefined) {
+      if (
+        failure.counterexample !== undefined &&
+        failure.counterexample !== null
+      ) {
         const counterexamples = Array.isArray(failure.counterexample)
           ? failure.counterexample
           : [failure.counterexample];
 
         for (const counterexample of counterexamples) {
-          const details = `{\n${Object.entries(counterexample)
```

Comment:
I removed the `{}` enclosing from the counterexample as it didn't provide any additional info but took up valuable space. Not having it also make it more consistent with the rest. See the example output in the description.